mitodl/ocw-studio
# use new site pipeline definition

# What are the relevant tickets?
https://github.com/mitodl/ocw-studio/issues/1928

# Description (What does it do?)
This PR replaces the use of the static `site-pipeline.yml` definition with the new `SitePipelineDefinition` object based pipeline.

# How can this be tested?
 - Spin up `ocw-studio`, making sure you have Minio and a custom Github org configured and have run the theme assets pipeline at least once (`docker-compose exec web ./manage.py upsert_theme_assets_pipeline`)
 - Pick a course to test and note its `short_id` property
 - Run `docker-compose exec web ./manage.py backpopulate_pipelines --filter short_id`, replacing `short_id` with the ID of your site
 - In the `ocw-studio` UI, publish any site
 - Open the Concourse UI at http://localhost:8080/ and find the pipeline for your test site
 - Watch the pipeline run, inspecting each step as it runs and make sure that everything looks correct:
   - Steps being run in the proper order
   - Files being synced appropriately to the right places
   - No errors
 - Back in the `ocw-studio` UI, click the link in the publish drawer to browse the deployed site and verify that everything looks good
